lumicks/pylake/kymo.py

```diff
@@ -150,6 +150,87 @@ def set_aspect_ratio(axis, ar):
         set_aspect_ratio(ax1, aspect_ratio)
         set_aspect_ratio(ax2, aspect_ratio)
 
+    def _histogram_rows(self, image, pixels_per_bin, pixel_width):
+        bin_width = pixels_per_bin * pixel_width # in physical units
+        n_rows = image.shape[0]
+        if pixels_per_bin > n_rows:
+            raise ValueError("bin size is larger than the available pixels")
+
+        n_bins = n_rows // pixels_per_bin
+        remainder = n_rows % pixels_per_bin
+        if remainder != 0:
+            warnings.warn(f"{n_rows} pixels is not divisible by {n_bins} bins, skipping last {remainder} pixels")
+        image = image[:(n_bins * pixels_per_bin)].reshape((n_bins, -1))
```

Comment:
I find the skipping for the overhanging pixel a strange choice from a user perspective. The warning is good, but seeing the last pixel be empty is strange to me when I have the data right there. I don't feel that strongly about it, but thought I'd just mention it.

![image](https://user-images.githubusercontent.com/19836026/101931022-2fc03600-3bd9-11eb-8f7e-0c909e1045cf.png)